modules/httpcache/httpcache.go

```diff
@@ -40,21 +42,27 @@ func HandleTimeCache(req *http.Request, w http.ResponseWriter, fi os.FileInfo) (
 		}
 	}
 
-	w.Header().Set("Cache-Control", GetCacheControl())
 	w.Header().Set("Last-Modified", fi.ModTime().Format(http.TimeFormat))
 	return false
 }
 
-// HandleEtagCache handles ETag-based caching for a HTTP request
-func HandleEtagCache(req *http.Request, w http.ResponseWriter, fi os.FileInfo) (handled bool) {
+// HandleFileEtagCache handles ETag-based caching for a HTTP request
+func HandleFileEtagCache(req *http.Request, w http.ResponseWriter, fi os.FileInfo) (handled bool) {
+	w.Header().Set("Cache-Control", GetCacheControl())
+	
 	etag := generateETag(fi)
-	if req.Header.Get("If-None-Match") == etag {
-		w.Header().Set("ETag", etag)
-		w.WriteHeader(http.StatusNotModified)
-		return true
-	}
+	return HandleGenericETagCache(req, w, etag)
+}
 
-	w.Header().Set("Cache-Control", GetCacheControl())
-	w.Header().Set("ETag", etag)
+// HandleGenericETagCache handles ETag-based caching for a HTTP request.
+// It returns true if the request was handled.
+func HandleGenericETagCache(req *http.Request, w http.ResponseWriter, etag string) bool {
+	if len(etag) > 0 {
+		w.Header().Set("Etag", etag)
+		if req.Header.Get("If-None-Match") == etag {
```

Comment:
Just saw on [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match) that `If-None-Match` can have have multiple values:

```
If-None-Match: "<etag_value>"
If-None-Match: "<etag_value>", "<etag_value>", …
```
Maybe we want to support that as well here.